### ISSUE TITLE
panic if number of open file descriptors exceeds the limit (due to multiple AVRO OCF sinks)

### ISSUE THREAD

**Issue by philip-stoev** [MaterializeInc/materialize#7132]
### What version of Materialize are you using?

```
materialized v0.8.1-dev (881c725fd)

```

### What was the issue?

In https://github.com/benesch/materialize/commit/8b3b192541f51db9b7a0db4ee60b7d495d29a383 an attempt was made to increase the number of available file descriptors to the hard limit. However, it does not protect against cases where the hard limit is low, such as 1024.

With a big number of workers and e.g. AVRO OCF sinks, each worker will have a file descriptor open for every sink. Materialize will then crash with:

```
materialized v0.8.1-dev (881c725fd) listening on 0.0.0.0:6875...
materialized encountered an internal error and crashed.

We rely on bug reports to diagnose and fix these errors. Please
copy and paste the above details and file a report at:

    https://materialize.com/s/bug
```

and no other information is provided. The stack trace winds all the way through the product:

```
(gdb) bt
#0  core::sync::atomic::atomic_load () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/core/src/sync/atomic.rs:2352
#1  core::sync::atomic::AtomicUsize::load () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/core/src/sync/atomic.rs:1491
#2  std::sync::once::Once::is_completed () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/std/src/sync/once.rs:370
#3  std::sync::once::Once::call_once () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/std/src/sync/once.rs:256
#4  lazy_static::lazy::Lazy<T>::get () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/lazy_static-1.4.0/src/inline_lazy.rs:30
#5  <materialized::PANIC_MUTEX as core::ops::deref::Deref>::deref::__stability () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/lazy_static-1.4.0/src/lib.rs:142
#6  <materialized::PANIC_MUTEX as core::ops::deref::Deref>::deref () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/lazy_static-1.4.0/src/lib.rs:144
#7  materialized::handle_panic () at /home/pstoev/materialize/src/materialized/src/bin/materialized/main.rs:651
#8  0x0000555556854d69 in core::ops::function::Fn::call () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/core/src/ops/function.rs:70
#9  0x0000555558d88d66 in std::panicking::rust_panic_with_hook () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0//library/std/src/panicking.rs:595
#10 0x0000555558d88887 in std::panicking::begin_panic_handler::{{closure}} () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0//library/std/src/panicking.rs:497
#11 0x0000555558d8670c in std::sys_common::backtrace::__rust_end_short_backtrace () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0//library/std/src/sys_common/backtrace.rs:141
#12 0x0000555558d887e9 in rust_begin_unwind () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0//library/std/src/panicking.rs:493
#13 0x0000555558daa231 in core::panicking::panic_fmt () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0//library/core/src/panicking.rs:92
#14 0x0000555558daa053 in core::option::expect_none_failed () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0//library/core/src/option.rs:1300
#15 0x000055555675a114 in core::result::Result<T,E>::expect () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/core/src/result.rs:997
#16 materialized::run::{{closure}} () at /home/pstoev/materialize/src/materialized/src/bin/materialized/main.rs:486
#17 <F as tracing_subscriber::fmt::writer::MakeWriter>::make_writer () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-subscriber-0.2.18/src/fmt/writer.rs:54
#18 <tracing_subscriber::fmt::fmt_layer::Layer<S,N,E,W> as tracing_subscriber::layer::Layer<S>>::on_event::{{closure}} ()
    at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-subscriber-0.2.18/src/fmt/fmt_layer.rs:743
#19 std::thread::local::LocalKey<T>::try_with () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/std/src/thread/local.rs:272
#20 std::thread::local::LocalKey<T>::with () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/std/src/thread/local.rs:248
#21 0x0000555556814747 in <tracing_subscriber::fmt::fmt_layer::Layer<S,N,E,W> as tracing_subscriber::layer::Layer<S>>::on_event ()
    at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-subscriber-0.2.18/src/fmt/fmt_layer.rs:726
#22 <tracing_subscriber::layer::Layered<L,S> as tracing_core::subscriber::Subscriber>::event () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-subscriber-0.2.18/src/layer.rs:651
#23 <tracing_subscriber::layer::Layered<L,S> as tracing_core::subscriber::Subscriber>::event () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-subscriber-0.2.18/src/layer.rs:650
#24 0x00005555589d17c9 in <alloc::sync::Arc<dyn tracing_core::subscriber::Subscriber+core::marker::Send+core::marker::Sync> as tracing_core::subscriber::Subscriber>::event ()
    at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-core-0.1.18/src/subscriber.rs:677
#25 tracing_core::dispatcher::Dispatch::event () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-core-0.1.18/src/dispatcher.rs:532
#26 tracing_log::dispatch_record::{{closure}} () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-log-0.1.2/src/lib.rs:204
#27 0x00005555589d13bd in tracing_core::dispatcher::get_default::{{closure}} () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-core-0.1.18/src/dispatcher.rs:346
#28 std::thread::local::LocalKey<T>::try_with () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/std/src/thread/local.rs:272
#29 tracing_core::dispatcher::get_default () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-core-0.1.18/src/dispatcher.rs:343
#30 0x00005555589cf6fd in tracing_log::dispatch_record () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-log-0.1.2/src/lib.rs:188
#31 <tracing_log::log_tracer::LogTracer as log::Log>::log () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-log-0.1.2/src/log_tracer.rs:186
#32 0x0000555558ce22d6 in log::__private_api_log () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/log-0.4.14/src/lib.rs:1460
#33 0x00005555571ac92b in rdkafka::client::ClientContext::log () at /home/pstoev/.cargo/git/checkouts/rust-rdkafka-34d944143d921bb0/b4ec50e/src/client.rs:71
#34 0x0000555557c25011 in rdkafka::client::native_log_cb () at /home/pstoev/.cargo/git/checkouts/rust-rdkafka-34d944143d921bb0/b4ec50e/src/client.rs:387
#35 0x000055555819a30d in rd_kafka_log_buf (
    buf=0x7ffd211e5ce0 "[thrd:GroupCoordinator]: GroupCoordinator: Ubuntu-2004-focal-64-minimal:9092: Failed to resolve 'Ubuntu-2004-focal-64-minimal:9092': Too many open files (after 0ms in state CONNECT)", fac=0x555555d561a6 "FAIL", ctx=0, level=3, rk=0x7fffe5820000, conf=0x7fffe5820138) at /home/pstoev/.cargo/git/checkouts/rust-rdkafka-34d944143d921bb0/b4ec50e/rdkafka-sys/librdkafka/src/rdkafka.c:253
#36 rd_kafka_log0 (conf=0x7fffe5820138, rk=0x7fffe5820000, extra=extra@entry=0x0, level=level@entry=3, ctx=ctx@entry=0, fac=fac@entry=0x555555d561a6 "FAIL", fmt=0x555555d422ee "%s: %s")
    at /home/pstoev/.cargo/git/checkouts/rust-rdkafka-34d944143d921bb0/b4ec50e/rdkafka-sys/librdkafka/src/rdkafka.c:293
#37 0x00005555581a8d35 in rd_kafka_broker_set_error (rkb=rkb@entry=0x7ffdc5529000, level=level@entry=3, err=err@entry=RD_KAFKA_RESP_ERR__RESOLVE, fmt=fmt@entry=0x555555d5838c "Failed to resolve '%s': %s", 
    ap=ap@entry=0x7ffd211e6a58) at /home/pstoev/.cargo/git/checkouts/rust-rdkafka-34d944143d921bb0/b4ec50e/rdkafka-sys/librdkafka/src/rdkafka_broker.c:500
#38 0x00005555581aaeaa in rd_kafka_broker_fail (rkb=rkb@entry=0x7ffdc5529000, level=level@entry=3, err=err@entry=RD_KAFKA_RESP_ERR__RESOLVE, fmt=fmt@entry=0x555555d5838c "Failed to resolve '%s': %s")
    at /home/pstoev/.cargo/git/checkouts/rust-rdkafka-34d944143d921bb0/b4ec50e/rdkafka-sys/librdkafka/src/rdkafka_broker.c:545
#39 0x00005555581bf3e0 in rd_kafka_broker_resolve (nodename=0x7ffd211e6cb0 "Ubuntu-2004-focal-64-minimal:9092", rkb=0x7ffdc5529000)
    at /home/pstoev/.cargo/git/checkouts/rust-rdkafka-34d944143d921bb0/b4ec50e/rdkafka-sys/librdkafka/src/rdkafka_broker.c:961
#40 rd_kafka_broker_connect (rkb=0x7ffdc5529000) at /home/pstoev/.cargo/git/checkouts/rust-rdkafka-34d944143d921bb0/b4ec50e/rdkafka-sys/librdkafka/src/rdkafka_broker.c:2103
#41 rd_kafka_broker_thread_main (arg=0x7ffdc5529000) at /home/pstoev/.cargo/git/checkouts/rust-rdkafka-34d944143d921bb0/b4ec50e/rdkafka-sys/librdkafka/src/rdkafka_broker.c:5193
#42 0x00007ffff7fa56da in start_thread (arg=<optimized out>) at pthread_create.c:474
#43 0x00007ffff7d40293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
(gdb) frame 16
#16 materialized::run::{{closure}} () at /home/pstoev/materialize/src/materialized/src/bin/materialized/main.rs:486
486	                            file.try_clone().expect("failed to clone log file")
(gdb) list
481	                            .append(true)
482	                            .create(true)
483	                            .open(&path)
484	                            .with_context(|| format!("creating log file: {}", path.display()))?;
485	                        fmt::layer().with_ansi(false).with_writer(move || {
486	                            file.try_clone().expect("failed to clone log file")
487	                        })
488	                    })
489	                    .with(FilterLayer::new(
490	                        fmt::layer()
(gdb) 
```
or even:

```
(gdb) bt
#0  core::sync::atomic::atomic_load () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/core/src/sync/atomic.rs:2352
#1  core::sync::atomic::AtomicUsize::load () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/core/src/sync/atomic.rs:1491
#2  std::sync::once::Once::is_completed () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/std/src/sync/once.rs:370
#3  std::sync::once::Once::call_once () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/std/src/sync/once.rs:256
#4  lazy_static::lazy::Lazy<T>::get () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/lazy_static-1.4.0/src/inline_lazy.rs:30
#5  <materialized::PANIC_MUTEX as core::ops::deref::Deref>::deref::__stability () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/lazy_static-1.4.0/src/lib.rs:142
#6  <materialized::PANIC_MUTEX as core::ops::deref::Deref>::deref () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/lazy_static-1.4.0/src/lib.rs:144
#7  materialized::handle_panic () at /home/pstoev/materialize/src/materialized/src/bin/materialized/main.rs:651
#8  0x0000555556854d69 in core::ops::function::Fn::call () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/core/src/ops/function.rs:70
#9  0x0000555558d88d66 in std::panicking::rust_panic_with_hook () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0//library/std/src/panicking.rs:595
#10 0x0000555558d88887 in std::panicking::begin_panic_handler::{{closure}} () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0//library/std/src/panicking.rs:497
#11 0x0000555558d8670c in std::sys_common::backtrace::__rust_end_short_backtrace () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0//library/std/src/sys_common/backtrace.rs:141
#12 0x0000555558d887e9 in rust_begin_unwind () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0//library/std/src/panicking.rs:493
#13 0x0000555558daa231 in core::panicking::panic_fmt () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0//library/core/src/panicking.rs:92
#14 0x0000555558daa053 in core::option::expect_none_failed () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0//library/core/src/option.rs:1300
#15 0x000055555675a114 in core::result::Result<T,E>::expect () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/core/src/result.rs:997
#16 materialized::run::{{closure}} () at /home/pstoev/materialize/src/materialized/src/bin/materialized/main.rs:486
#17 <F as tracing_subscriber::fmt::writer::MakeWriter>::make_writer () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-subscriber-0.2.18/src/fmt/writer.rs:54
#18 <tracing_subscriber::fmt::fmt_layer::Layer<S,N,E,W> as tracing_subscriber::layer::Layer<S>>::on_event::{{closure}} ()
    at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-subscriber-0.2.18/src/fmt/fmt_layer.rs:743
#19 std::thread::local::LocalKey<T>::try_with () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/std/src/thread/local.rs:272
#20 std::thread::local::LocalKey<T>::with () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/std/src/thread/local.rs:248
#21 0x0000555556814747 in <tracing_subscriber::fmt::fmt_layer::Layer<S,N,E,W> as tracing_subscriber::layer::Layer<S>>::on_event ()
    at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-subscriber-0.2.18/src/fmt/fmt_layer.rs:726
#22 <tracing_subscriber::layer::Layered<L,S> as tracing_core::subscriber::Subscriber>::event () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-subscriber-0.2.18/src/layer.rs:651
#23 <tracing_subscriber::layer::Layered<L,S> as tracing_core::subscriber::Subscriber>::event () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-subscriber-0.2.18/src/layer.rs:650
#24 0x00005555589d17c9 in <alloc::sync::Arc<dyn tracing_core::subscriber::Subscriber+core::marker::Send+core::marker::Sync> as tracing_core::subscriber::Subscriber>::event ()
    at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-core-0.1.18/src/subscriber.rs:677
#25 tracing_core::dispatcher::Dispatch::event () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-core-0.1.18/src/dispatcher.rs:532
#26 tracing_log::dispatch_record::{{closure}} () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-log-0.1.2/src/lib.rs:204
#27 0x00005555589d13bd in tracing_core::dispatcher::get_default::{{closure}} () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-core-0.1.18/src/dispatcher.rs:346
#28 std::thread::local::LocalKey<T>::try_with () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/std/src/thread/local.rs:272
#29 tracing_core::dispatcher::get_default () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-core-0.1.18/src/dispatcher.rs:343
#30 0x00005555589cf6fd in tracing_log::dispatch_record () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-log-0.1.2/src/lib.rs:188
#31 <tracing_log::log_tracer::LogTracer as log::Log>::log () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-log-0.1.2/src/log_tracer.rs:186
#32 0x0000555558ce22d6 in log::__private_api_log () at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/log-0.4.14/src/lib.rs:1460
#33 0x00005555571b8024 in dataflow::sink::avro_ocf::avro_ocf () at /home/pstoev/materialize/src/dataflow/src/sink/avro_ocf.rs:50
#34 0x00005555577fbbcc in dataflow::render::sinks::<impl dataflow::render::context::Context<timely::dataflow::scopes::child::Child<G,<G as timely::dataflow::scopes::ScopeParent>::Timestamp>,repr::row::Row,u64>>::export_sink () at /home/pstoev/materialize/src/dataflow/src/render/sinks.rs:262
#35 0x0000555557e1d002 in dataflow::render::build_dataflow::{{closure}}::{{closure}} () at /home/pstoev/materialize/src/dataflow/src/render/mod.rs:236
#36 <timely::dataflow::scopes::child::Child<G,T> as timely::dataflow::scopes::Scope>::scoped ()
    at /home/pstoev/.cargo/git/checkouts/timely-dataflow-4c0cc365061cd263/d14afd6/timely/src/dataflow/scopes/child.rs:129
#37 timely::dataflow::scopes::Scope::region_named () at /home/pstoev/.cargo/git/checkouts/timely-dataflow-4c0cc365061cd263/d14afd6/timely/src/dataflow/scopes/mod.rs:187
#38 timely::dataflow::scopes::Scope::region () at /home/pstoev/.cargo/git/checkouts/timely-dataflow-4c0cc365061cd263/d14afd6/timely/src/dataflow/scopes/mod.rs:155
#39 0x00005555577b987d in dataflow::render::build_dataflow::{{closure}} () at /home/pstoev/materialize/src/dataflow/src/render/mod.rs:191
#40 timely::worker::Worker<A>::dataflow_core () at /home/pstoev/.cargo/git/checkouts/timely-dataflow-4c0cc365061cd263/d14afd6/timely/src/worker.rs:639
#41 0x0000555557ea6cd6 in dataflow::render::build_dataflow () at /home/pstoev/materialize/src/dataflow/src/render/mod.rs:186
#42 0x00005555578084ce in dataflow::server::Worker<A>::handle_command () at /home/pstoev/materialize/src/dataflow/src/server.rs:669
#43 0x00005555577bdd84 in dataflow::server::Worker<A>::run () at /home/pstoev/materialize/src/dataflow/src/server.rs:465
#44 dataflow::server::serve::{{closure}} () at /home/pstoev/materialize/src/dataflow/src/server.rs:232
#45 timely::execute::execute::{{closure}} () at /home/pstoev/.cargo/git/checkouts/timely-dataflow-4c0cc365061cd263/d14afd6/timely/src/execute.rs:285
#46 0x000055555775a0d0 in timely_communication::initialize::initialize_from::{{closure}} () at /home/pstoev/.cargo/git/checkouts/timely-dataflow-4c0cc365061cd263/d14afd6/communication/src/initialize.rs:285
#47 std::sys_common::backtrace::__rust_begin_short_backtrace () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/std/src/sys_common/backtrace.rs:125
#48 0x000055555775ce2d in std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}} () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/std/src/thread/mod.rs:474
#49 <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/std/src/panic.rs:344
#50 std::panicking::try::do_call () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/std/src/panicking.rs:379
#51 std::panicking::try () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/std/src/panicking.rs:343
#52 std::panic::catch_unwind () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/std/src/panic.rs:431
#53 std::thread::Builder::spawn_unchecked::{{closure}} () at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/std/src/thread/mod.rs:473
```

### Is the issue reproducible? If so, please provide reproduction instructions.

1. ```set ulimit -c 1024``` 
2. run materialize with 64 workers
3. run this testdrive:
```
$ set upsert-keyschema={
    "type": "record",
    "name": "Key",
    "fields": [
        {"name": "key1", "type": "string"},
        {"name": "key2", "type": "long"}
    ]
  }

$ set upsert-schema={
        "type" : "record",
        "name" : "test",
        "fields" : [
            {"name":"f1", "type":"string"},
            {"name":"f2", "type":"long"}
        ]
    }

$ kafka-create-topic topic=upsert-avro

$ kafka-ingest format=avro topic=upsert-avro key-format=avro key-schema=${upsert-keyschema} schema=${upsert-schema} publish=true
{"key1": "fish", "key2": 2} {"f1": "fish", "f2": 1000}
{"key1": "fisch", "key2": 42} {"f1": "fish", "f2": 1000}
{"key1": "fish", "key2": 21} {"f1": "fish", "f2": 1000}
{"key1": "fisch", "key2": 142} {"f1": "fish", "f2": 1000}
{"key1": "fish", "key2": 22} {"f1": "fish", "f2": 1000}
{"key1": "fisch", "key2": 242} {"f1": "fish", "f2": 1000}
{"key1": "fish", "key2": 32} {"f1": "fish", "f2": 1000}
{"key1": "fisch", "key2": 342} {"f1": "fish", "f2": 1000}
{"key1": "fish", "key2": 42} {"f1": "fish", "f2": 1000}
{"key1": "fisch", "key2": 442} {"f1": "fish", "f2": 1000}
{"key1": "fish", "key2": 52} {"f1": "fish", "f2": 1000}
{"key1": "fisch", "key2": 542} {"f1": "fish", "f2": 1000}
{"key1": "fish", "key2": 62} {"f1": "fish", "f2": 1000}
{"key1": "fisch", "key2": 642} {"f1": "fish", "f2": 1000}
{"key1": "fish", "key2": 72} {"f1": "fish", "f2": 1000}
{"key1": "fisch", "key2": 742} {"f1": "fish", "f2": 1000}

> CREATE MATERIALIZED SOURCE upsert_input
  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC
  'testdrive-upsert-avro-${testdrive.seed}'
  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}'
  ENVELOPE UPSERT

> CREATE SINK upsert_input_sink0 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink1 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink2 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink3 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink4 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink5 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink6 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink7 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink8 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink9 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink10 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink11 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink12 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink13 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink14 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink15 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink16 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink17 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink18 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink19 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink20 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink21 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink22 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink23 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink25 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
> CREATE SINK upsert_input_sink26 FROM upsert_input INTO AVRO OCF '/tmp/foo.ocf';
```

**Comment by philip-stoev** [comment]
This was previously discussed as #1919, and the discussion there turned to kafka sources.

In this ticket here, the reproduction scenario uses AVRO OCF sinks.

**Comment by aljoscha** [comment]
I talked to @philip-stoev and he mentioned he's fine  with closing this because of the fix in #7245.
